orchagent/intfsorch.cpp

```diff
@@ -671,3 +715,104 @@ void IntfsOrch::removeDirectedBroadcast(const Port &port, const IpAddress &ip_ad
 
     SWSS_LOG_NOTICE("Remove broadcast route ip:%s", ip_addr.to_string().c_str());
 }
+
+void IntfsOrch::addRifToFlexCounter(const string &id, const string &name, const string &type)
+{
+    SWSS_LOG_ENTER();
+    /* update RIF maps in COUNTERS_DB */
+    vector<FieldValueTuple> rifNameVector;
+    vector<FieldValueTuple> rifTypeVector;
+
+    rifNameVector.emplace_back(name, id);
+    rifTypeVector.emplace_back(id, type);
+
+    m_rifNameTable->set("", rifNameVector);
+    m_rifTypeTable->set("", rifTypeVector);
+
+    /* update RIF in FLEX_COUNTER_DB */
+    string key = getRifFlexCounterTableKey(id);
+
+
+    std::ostringstream counters_stream;
+    for (const auto& it: rifStatIds)
+    {
+        counters_stream << sai_serialize_router_interface_stat(it) << comma;
+    }
+
+    /* check the state of intf, if registering the intf to FC will result in runtime error */
+    vector<FieldValueTuple> fvt;
+    vector<FieldValueTuple> fieldValues;
+    fieldValues.emplace_back(RIF_COUNTER_ID_LIST, counters_stream.str());
+
+    m_flexCounterTable->set(key, fieldValues);
+    SWSS_LOG_DEBUG("Registered interface %s to Flex counter", name.c_str());
+}
+
+void IntfsOrch::removeRifFromFlexCounter(const string &id, const string &name)
+{
+    SWSS_LOG_ENTER();
+    /* remove it from COUNTERS_DB maps */
+    m_rifNameTable->hdel("", name);
+    m_rifTypeTable->hdel("", id);
+
+    /* remove it from FLEX_COUNTER_DB */
+    string key = getRifFlexCounterTableKey(id);
+
+    vector<FieldValueTuple> fieldValues;
+    fieldValues.emplace_back(RIF_COUNTER_ID_LIST, "");
+
+    m_flexCounterTable->set(key, fieldValues);
+    SWSS_LOG_DEBUG("Unregistered interface %s from Flex counter", name.c_str());
+}
+
+string IntfsOrch::getRifFlexCounterTableKey(string key)
+{
+    return string(RIF_STAT_COUNTER_FLEX_COUNTER_GROUP) + ":" + key;
+}
+
+void IntfsOrch::generateInterfaceMap()
+{
+    m_updateMapsTimer->start();
+}
+
+void IntfsOrch::doTask(SelectableTimer &timer)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_NOTICE("Registering %ld new intfs, deleting %ld ", m_rifsToAdd.size(), m_rifsToRemove.size());
+    string value;
+    for (auto it = m_rifsToAdd.begin(); it != m_rifsToAdd.end(); ++it)
+    {
+        SWSS_LOG_NOTICE("Registering begin -> %s ", it->m_alias.c_str());
```

Comment:
Why do we have a separate for-loop for this log? With the log in below for-loop, this appears to be redundant.